feature: ZENKO-1383 extend lifecycle transition tests

jonathan-gramain opened the pull request:
**What does this PR do, and why do we need it?**

feature: ZENKO-1383 extend lifecycle transition tests
    
- test transition from/to all major clouds AWS, GCP, Azure

- add an MPU transition for each test


**Which issue does this PR fix?**

ZENKO-1383

**Special notes for your reviewers**:

Thank you :heart:

jonathan-gramain committed:
feature: ZENKO-1383 enable and extend lifecycle transition tests

- update docker-entrypoint.sh to run lifecycle transition E2E tests.

- test transition from/to all major clouds AWS, GCP, Azure

- add an MPU transition for each test

- create topic "backbeat-data-mover" in CI environment

- shorter interval for lifecycle condutor to speed up testing

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

ghost commented:
There were the following issues with this Pull Request

* Commit: e450e5d79522f759eb3227d3533c8d1af9e8aa69
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @ssalaues

* @nicolast

* @vrancurel

bennettbuchanan commented on tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js:
```diff
@@ -23,32 +22,110 @@ class LifecycleUtility extends ReplicationUtility {
+
+    createBucket(cb) {
+        if (this.sourceLocation) {
+            this.s3.createBucket({
```
Minor: if we check `this.sourceLocation` to be 'Zenko', then we can create the bucket and return. That way we can avoid the else block.

jonathan-gramain replied:
I think it would have to be "us-east-1", because that must be the name of the "standard" zenko location. But it can work if we use 'Zenko' for the location family and set the "name" field to be "us-east-1".

bennettbuchanan commented on tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js:
```diff
@@ -23,32 +22,110 @@ class LifecycleUtility extends ReplicationUtility {
+                    Status: 'Enabled',
+                }
+            }, next),
+        ], err => cb(err));
```
Minor: can just pass `cb` here.

jonathan-gramain replied:
Possibly, I'm just wary because AFAIK `async.series` returns an argument to its callback with the individual results, and it could confuse constructs using `async.waterfall` or so calling `createVersionedBucket()`, so may be better leaving an explicit call, just so that the callback contract is clear.

bennettbuchanan commented on tests/zenko_tests/node_tests/backbeat/tests/lifecycle/transition.js:
```diff
@@ -11,61 +13,122 @@ const keyPrefix = `${uuid()}-key-prefix`;
-        cloudServer._compareObjectBody(data[0].Body, data[3].Body);
+        // check data stored on the target cloud
+        cloudServer._compareObjectBody(data[0].Body, data[3]);
+        // check that object is still readable from Zenko
```
👍

bennettbuchanan commented on tests/zenko_tests/node_tests/backbeat/tests/lifecycle/transition.js:
```diff
@@ -11,61 +13,122 @@ const keyPrefix = `${uuid()}-key-prefix`;
-describe('transition to AWS backend', function() {
-    this.timeout(LIFECYCLE_TIMEOUT);
+const locationParams = {
+    local: {
```
Minor: since we're using the names of the public clouds, I think it reads a bit easier if this is 'Zenko'.

jonathan-gramain replied:
Yes, while setting the location name to "us-east-1", it could make code a bit simpler indeed.

bennettbuchanan commented on tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js:
```diff
@@ -23,32 +22,110 @@ class LifecycleUtility extends ReplicationUtility {
+        }
+    }
+
+    getMetadataGCP(cb) {
```
Looks like this is unused.

bennettbuchanan commented on tests/zenko_tests/node_tests/backbeat/tests/lifecycle/transition.js:
```diff
@@ -11,61 +13,122 @@ const keyPrefix = `${uuid()}-key-prefix`;
+}, {
+    from: 'Azure',
+    to: 'AWS',
+}].forEach(test => {
```
This is solid 🥇

bennettbuchanan commented on tests/zenko_tests/node_tests/backbeat/tests/lifecycle/transition.js:
```diff
@@ -11,61 +13,122 @@ const keyPrefix = `${uuid()}-key-prefix`;
-        beforeEach(done => cloudServer.createBucket(srcBucket, done));
+        before(() => {
+            if (test.from !== 'local') {
+                cloudServer.setSourceLocation(fromLoc.name);
```
I think we should always set this (i.e. also in the 'Zenko' case), and then let the methods handle any logic related to the location type. That way we can benefit more from method chaining.

bennettbuchanan reviewed (approved):
Nice additions! Overall LGTM. Left some ideas around the design of the helper utilities for readability.

ghost commented:
There were the following issues with this Pull Request

* Commit: 530b0355b186debcbef9fab0d001b6305774fe4f
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 530b0355b186debcbef9fab0d001b6305774fe4f
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 7feb6113ecbb99c4cd9612cea927321a94aa092c
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 530b0355b186debcbef9fab0d001b6305774fe4f
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 7feb6113ecbb99c4cd9612cea927321a94aa092c
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: c31cb92e668eb880ba998435f42249348b5a2c01
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: e0a47b79940ca1d4f666fd8bacdfe0743712d957
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 9e41750863e436cd1ce4b01d5db80beab3865fff
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-1383 contains:


* `1.1.0`


Considering where you are trying to merge, I expected to find:


* `1.1.0`

* `1.2.0`


Please check the `Fix Version/s` of ZENKO-1383, or the target
branch of this pull request.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/ZENKO-1383/transition-integration-tests-extra` into
`development/1.1`
* [w/1.2/feature/ZENKO-1383/transition-integration-tests-extra](https://github.com/scality/zenko/compare/development/1.2...w/1.2/feature/ZENKO-1383/transition-integration-tests-extra) will be merged into `development/1.2` (pull request #603)



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @ssalaues

* @nicolast

* @vrancurel

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-1383 contains:


* `1.1.0`

* `1.2.0`


Considering where you are trying to merge, I expected to find:


* `1.1.0`


Please check the `Fix Version/s` of ZENKO-1383, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @ssalaues

* @nicolast

* @vrancurel

ssalaues commented:
Tests that this PR covers passed in CI, bypassing the build status for a known bug

jonathan-gramain commented:
@bert-e approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/Zenko/commit/63bfbe24e1315ebc1f22d4b457865df6d24b255f) did not succeed in branch feature/ZENKO-1383/transition-integration-tests-extra.




*The following options are set:* **approve**

ssalaues commented:
@bert-e bypass_build_status

bert-e commented:
# Conflict with a changeset in the queue



The changeset in this pull request conflicts with another changeset
already in the queue. Please wait for the current queue to merge into
the development branch. The conflict will then appear in this pull
request and can be sorted on the feature branch directly.

This changeset has *not* been added to the queue. 



*The following options are set:* **bypass_build_status, approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.1`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **bypass_build_status, approve**

jonathan-gramain committed:
[workaround] S3C-1383: reduce parallelism on MPU putPart

Reduce the number of parts put in parallel from 10 to 2 in the
tests. That might help regarding the ECONNRESET we see in the AWS mock
logs that fails tests.

ghost commented:
There were the following issues with this Pull Request

* Commit: ecca4b727b08e3def16cd94238d6071176211e17
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

bert-e commented:
# Partial merge



Apparently, new commits were added to the source branch after I had queued this
Pull Request. As a result, I have merged only **part of** it into the targetted
development branches:


* :heavy_check_mark: `development/1.1`


I will now work on the following commits:


* #ecca4b727b08e3def16cd94238d6071176211e17


**Please avoid this situation next time.**

As a friendly reminder, you can use the `wait` command to prevent me from processing your
pull-request while your work is still in progress.

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.1`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **bypass_build_status, approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.1`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`



Please check the status of the associated issue ZENKO-1383.

Goodbye jonathan-gramain.
